chaindataloader/binary.go

```diff
@@ -0,0 +1,227 @@
+package chaindataloader
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/neutrino/headerfs"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+/*
+
+================================================================================
+
+ENCODING FORMAT
+================================================================================
+
+Each file has a header of varint size consisting of,
+
+dataType || startHeight || endHeight || chain
+
+in that order.
+*/
+
+// binReader is an internal struct that holds all data the binReader needs
+// to fetch headers.
+type binReader struct {
+	// reader represents the source to be read.
+	reader io.ReadSeeker
```

Comment:
I like the usage of `io.ReaderSeeker` here! It means anything else can then just implement that interface. Eg: we can read from an HTTP source and abstract that away from the caller here.